src/main/java/com/netflix/imflibrary/st2067_2/Composition.java

```diff
@@ -0,0 +1,1154 @@
+
+        String result = "";
+
+        try(InputStream inputStream = resourceByteRangeProvider.getByteRangeAsStream(0, resourceByteRangeProvider.getResourceSize()-1);)
+        {
+            DocumentBuilderFactory documentBuilderFactory = DocumentBuilderFactory.newInstance();
+            documentBuilderFactory.setNamespaceAware(true);
+            DocumentBuilder documentBuilder = documentBuilderFactory.newDocumentBuilder();
+            documentBuilder.setErrorHandler(new ErrorHandler() {
+                @Override
+                public void warning(SAXParseException exception) throws SAXException {
+                    imfErrorLogger.addError(new ErrorLogger.ErrorObject(IMFErrorLogger.IMFErrors.ErrorCodes.IMF_CPL_ERROR, IMFErrorLogger.IMFErrors.ErrorLevels.WARNING, exception.getMessage()));
+                }
+
+                @Override
+                public void error(SAXParseException exception) throws SAXException {
+                    imfErrorLogger.addError(new ErrorLogger.ErrorObject(IMFErrorLogger.IMFErrors.ErrorCodes.IMF_CPL_ERROR, IMFErrorLogger.IMFErrors.ErrorLevels.NON_FATAL, exception.getMessage()));
+                }
+
+                @Override
+                public void fatalError(SAXParseException exception) throws SAXException {
+                    imfErrorLogger.addError(new ErrorLogger.ErrorObject(IMFErrorLogger.IMFErrors.ErrorCodes.IMF_CPL_ERROR, IMFErrorLogger.IMFErrors.ErrorLevels.FATAL, exception.getMessage()));
+                }
+            });
+            Document document = documentBuilder.parse(inputStream);
+
+            //obtain root node
+            NodeList nodeList = null;
+            for(String cplNamespaceURI : Composition.supportedCPLSchemaURIs) {
+                nodeList = document.getElementsByTagNameNS(cplNamespaceURI, "CompositionPlaylist");
+                if (nodeList != null
+                        && nodeList.getLength() == 1)
+                {
+                    result = cplNamespaceURI;
+                    break;
+                }
+            }
+        }
+        catch(ParserConfigurationException | SAXException e)
+        {
+            throw new IMFException(String.format("Error occurred while trying to determine the Composition Playlist Namespace URI, invalid CPL document Error Message : %s", e.getMessage()));
+        }
+        if(result.isEmpty()) {
+            throw new IMFException(String.format("Please check the CPL document and namespace URI, currently we only support the following schema URIs %s", serializeCollectionToString(supportedCPLSchemaURIs)));
+        }
+        return result;
+    }
+
+    private final String getCPLNamespaceVersion(String namespaceURI){
+        String[] uriComponents = namespaceURI.split("/");
+        String namespaceVersion = uriComponents[uriComponents.length - 1];
+        return namespaceVersion;
+    }
+
+    private final String serializeCollectionToString(Collection<String> set){
```

Comment:
Does this method belong to another class perhaps in 'com.netflix.imflibrary.utils' package? Can it be genericized?